src/headless/yare_impl/headless.rs

```diff
@@ -118,31 +144,127 @@ impl<F: Fn(u32)> Headless<F> {
                         }
                         // self energize
                         if index == target {
-                            for star in &self.stars {
+                            for (i, star) in self.stars.iter().enumerate() {
                                 // check distance
+                                if star.pos.dist(source_spirit.pos) < ENERGIZE_RANGE {
```

Comment:
Where range checks are concerned, you should use `<=`. Also you can get around having to do the square root by comparing the distances squared. Like so:
```
star.pos.dist_squared(source_spirit.pos) <= ENERGIZE_RANGE * ENERGIZE_RANGE
```
This operation is so common in my code, that I make a method for it:
```
star.pos.in_range(source_spirit.pos, ENERGIZE_RANGE)
```

Reply:
Although I just checked and it looks like `Vec2::in_range` uses `<` instead of `<=`. That should be fixed.